include/greybus/greybus.h

```diff
@@ -107,7 +107,8 @@ struct gb_transport_backend {
 
 struct gb_bundle {
     int id;                     /* bundle ID */
-    struct device *dev;         /* primary bundle device */
+    unsigned int cport_start;   /* start cport */
```

Comment:
I would prefer to keep a dev to cport mapping in a central location, and also to keep accesses to that mapping via functions. It's just a bit safer.

Also, technically, CPort 0 is not required to be present, but I'm not sure the code in its current state allows for that, so that should probably be fixed in a separate ticket. That would be useful for, say, the GPIO testsuite - it doesn't need to support the control CPort. 

The change to `struct device **` works for me though, but it needs a size associated with it.

Question: Is the `void *priv` used for anything at the moment? If not, that could be the size of the `struct device **`.

It's probably also quite possible that the `**dev` can be statically allocated in `DEFINE_GREYBUS_BUNDLE()`, using other DT macros to count the number of CPort children.

Ideally, it would also be compile-time const as well, but if necessary, the array of `struct device *` could also be populated in `greybus_bundle_init()`.

Reply:
Also, it might be possible to keep the `struct device *dev` and refer to a specific instance of `zephyr,greybus-bundle`. That way, we could create an API call to get a static array / size of `struct device *` that a particular bundle refers to.

Reply:
Thank you for the review @cfriedt  
```
The change to struct device ** works for me though, but it needs a size associated with it.

Question: Is the void *priv used for anything at the moment? If not, that could be the size of the struct device **
```
It looks like *priv is being used in a few places (pwm, sdio .etc)